tests/draft4/ref.json

```diff
@@ -141,6 +141,59 @@
         ]
     },
     {
+        "description": "recursive refs",
```

Comment:
This tests T4 from [the paper](http://cswr.github.io/JsonSchema/spec/why/), which checks that recsursive $refs are allowed. In this case I have made a schema that tests that implementations aren't thrown into an infinite loop, and can still validate parts of the schema which are not infinite.